x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -585,17 +586,38 @@ private Aggregate resolveAggregate(Aggregate aggregate, List<Attribute> children
 
                 List<NamedExpression> newAggregates = new ArrayList<>();
-                for (NamedExpression ag : aggregate.aggregates()) {
-                    var agg = (NamedExpression) ag.transformUp(UnresolvedAttribute.class, ua -> {
-                        Expression ne = ua;
-                        Attribute maybeResolved = maybeResolveAttribute(ua, resolvedList);
-                        if (maybeResolved != null) {
-                            changed.set(true);
-                            ne = maybeResolved;
-                        }
-                        return ne;
-                    });
-                    newAggregates.add(agg);
+                // If the groupings are not resolved, skip the resolution of the references to groupings in the aggregates, resolve the
+                // aggregations that do not reference to groupings, so that the fields/attributes referenced by the aggregations can be
+                // resolved, and verifier doesn't report field/reference/column not found errors for them.
+                boolean groupingResolved = Resolvables.resolved(groupings);
+                int size = groupingResolved ? aggregates.size() : aggregates.size() - groupings.size();
+                for (int i = 0; i < aggregates.size(); i++) {
+                    NamedExpression ag = aggregates.get(i);
+                    if (i < size) {
+                        var agg = (NamedExpression) ag.transformUp(UnresolvedAttribute.class, ua -> {
+                            Expression ne = ua;
+                            Attribute maybeResolved = maybeResolveAttribute(ua, resolvedList);
+                            if (groupingResolved) {
+                                if (maybeResolved != null) {
+                                    changed.set(true);
+                                    ne = maybeResolved;
+                                }
+                            } else {
+                                // An item in aggregations can reference to groupings explicitly, if groupings are not resolved yet and
+                                // maybeResolved is not resolved, return the original UnresolvedAttribute, so that it has a another chance
+                                // to get resolved in the next iteration.
+                                // For example STATS c = count(emp_no), x = d::int + 1 BY d = (date == "2025-01-01")
+                                if (maybeResolved instanceof Unresolvable == false) {
+                                    changed.set(true);
+                                    ne = maybeResolved;
+                                }
+                            }
+                            return ne;
+                        });
+                        newAggregates.add(agg);
+                    } else {
+                        newAggregates.add(ag); // Groupings are not resolved
+                    }
```

Comment:
Same here - instead of doing an if/else perform the condition only on updating agg and always update it similar to the previous code style:
```
var agg = ...
if (groupingResolved) {
   agg = 
}

newAggregates.add(agg);
```